x/market/keeper/keeper.go

```diff
@@ -106,8 +106,19 @@ func (k Keeper) CreateLease(ctx sdk.Context, bid types.Bid) {
 	}
 	key := leaseKey(lease.ID())
 
-	// XXX TODO: check not overwrite
+	// assert no overwrite
+	found := store.Get(key)
+	if found != nil { // value already exists
+		var val types.Lease
+		k.cdc.MustUnmarshalBinaryBare(found, &val)
+		k.updateActiveLeaseIndex(store, val)
+		return // do not overwrite
+	}
```

Comment:
let's not do this, the semantics are too confusing.  return an error if you want.

I think `CreateLease` is only used in the end blocker, where this case is checked already.

Reply:
Okay I removed it, left a comment. I didn't really see a great way to handle/enforce usage otherwise.